homeassistant/components/airtouch4/climate.py

```diff
@@ -0,0 +1,338 @@
+"""AirTouch 4 component to control of AirTouch 4 Climate Devices."""
+
+import logging
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    FAN_AUTO,
+    FAN_DIFFUSE,
+    FAN_FOCUS,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    HVAC_MODE_COOL,
+    HVAC_MODE_DRY,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import callback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from .const import DOMAIN
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+AT_TO_HA_STATE = {
+    "Heat": HVAC_MODE_HEAT,
+    "Cool": HVAC_MODE_COOL,
+    "AutoHeat": HVAC_MODE_AUTO,
+    "AutoCool": HVAC_MODE_AUTO,
+    "Auto": HVAC_MODE_AUTO,
+    "Dry": HVAC_MODE_DRY,
+    "Fan": HVAC_MODE_FAN_ONLY,
+    "Off": HVAC_MODE_OFF,
+}
+
+HA_STATE_TO_AT = {value: key for key, value in AT_TO_HA_STATE.items()}
+
+AT_TO_HA_FAN_SPEED = {
+    "Quiet": FAN_DIFFUSE,
+    "Low": FAN_LOW,
+    "Medium": FAN_MEDIUM,
+    "High": FAN_HIGH,
+    "Powerful": FAN_FOCUS,
+    "Auto": FAN_AUTO,
+    "Turbo": "turbo",
+}
+
+HA_FAN_SPEED_TO_AT = {value: key for key, value in AT_TO_HA_FAN_SPEED.items()}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _build_entity(coordinator, group_number, info):
+    _LOGGER.debug("Found device %s", group_number)
```

Comment:
Please just log the list of created entities once. The Entity class has a repr magic method.

Reply:
Hopefully I've gotten this right - if I've misunderstood let me know